extensions/wikia/AdEngine/js/MonetizationServiceHelper.js

```diff
@@ -0,0 +1,130 @@
+define('ext.wikia.adEngine.monetizationsServiceHelper', [
+	'jquery',
+	'wikia.geo',
+	'wikia.scriptwriter',
+	'wikia.tracker',
+	'wikia.window',
+], function ($, geo, scriptWriter, tracker, window) {
+	'use strict';
+
+	var isEndOfContent = false,
+		track = tracker.buildTrackingFunction({
+			trackingMethod: 'internal',
+			category: 'monetization-module',
+			geo: geo.getCountryCode()
+		});
+
+	function validateSlot(slotName) {
+		if (slotName === 'below_category' && isEndOfContent) {
+			return false;
+		}
+
+		return true;
+	}
+
+	function injectContent(slot, content, success) {
+		scriptWriter.injectHtml(slot, content, function () {
+			var $container = $('#' + slot).find('.monetization-module');
+
+			trackImpression($container);
+			trackClickEvent($container);
+
+			success();
+		});
+	}
+
+	function addInContentSlot(slot) {
+		var elementName = '#mw-content-text > h2',
+			num = $(elementName).length,
+			content = '<div id="' + slot + '" class="wikia-ad noprint default-height"></div>';
+
+		// Insert the ad above the 2nd <H2> tag. (Insert the ad above the 3rd <H2> tag if TOC exists)
+		if (num >= 2) {
+			$(elementName).eq(1).before(content);
+		// Otherwise, insert to the end of content
+		} else {
+			$('#mw-content-text').append(content);
+			isEndOfContent = true;
+		}
+
+		window.adslots2.push(slot);
+	}
+
+	function trackImpression($container) {
```

Comment:
For tracking we could merge this with GA ad tracking if that makes sense (it causes 1/100 sampling and is GA only, so maybe it's not what we want for Monetization Service?)